hazelcast/src/main/java/com/hazelcast/config/Config.java

```diff
@@ -311,12 +311,16 @@ public Config setMapConfigs(Map<String, MapConfig> mapConfigs) {
 
     public CacheSimpleConfig findCacheConfig(String name) {
         name = getBaseName(name);
-        return lookupByPattern(configPatternMatcher, cacheConfigs, name);
+        final CacheSimpleConfig config = lookupByPattern(configPatternMatcher, cacheConfigs, name);
```

Comment:
This change in `CacheSimpleConfig` lookup may alter our JCache implementation's behavior from throwing `CacheException` on `CacheManager.getCache(unknownCacheName)` to returning a `default`-configured `Cache`. There is already an issue for that (#10695) but it's a significant change and its consequences should be explored carefully, so I wouldn't do that in this PR.